pallets/token_migration/src/lib.rs

```diff
@@ -0,0 +1,230 @@
+    {
+        // Migrator transferred tokens
+        Migration(AccountId, AccountId, u128),
+
+        // Migrator's allowed migrations increased
+        MigratorExpanded(AccountId, u16),
+
+        // Migrator's allowed migrations decreased
+        MigratorContracted(AccountId, u16),
+    }
+);
+
+// The pallet's errors
+decl_error! {
+    /// Errors for the module.
+    pub enum Error for Module<T: Trait> {
+        MigratorAlreadyPresent,
+        UnknownMigrator,
+        ExceededMigrations,
+        CannotExpandMigrator,
+        CannotContractMigrator,
+        InsufficientBalance
+    }
+}
+
+decl_module! {
+    /// The module declaration.
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Does a token migration. The migrator should have sufficient balance to give tokens to recipients
+        /// The check whether it is a valid migrator is made inside the SignedExtension
+        // TODO: Set correct weight
+        #[weight = (T::DbWeight::get().reads_writes(1, recipients.len() as u64), Pays::No)]
+        pub fn migrate(origin, recipients: BTreeMap<T::AccountId, BalanceOf<T>>) -> dispatch::DispatchResult {
+            let migrator = ensure_signed(origin)?;
+            Self::migrate_(migrator, recipients)
+        }
+
+        /// Increase the migrators allowed migrations by the given number
+        #[weight = (0, Pays::No)]
+        pub fn expand_migrator(origin, migrator: T::AccountId, increase_migrations_by: u16) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            match Self::migrators(&migrator) {
+                Some(current_migrations) => {
+                    let new_migrations = current_migrations.checked_add(increase_migrations_by).ok_or(Error::<T>::CannotExpandMigrator)?;
+                    Migrators::<T>::insert(migrator.clone(), new_migrations.clone());
+                    Self::deposit_event(RawEvent::MigratorExpanded(migrator, new_migrations));
+                    Ok(())
+                },
+                None => fail!(Error::<T>::UnknownMigrator)
+            }
+        }
+
+        /// Decrease the migrators allowed migrations by the given number
+        #[weight = (0, Pays::No)]
+        pub fn contract_migrator(origin, migrator: T::AccountId, decrease_migrations_by: u16) -> dispatch::DispatchResult {
+            ensure_root(origin)?;
+            match Self::migrators(&migrator) {
+                Some(current_migrations) => {
+                    let new_migrations = current_migrations.checked_sub(decrease_migrations_by).ok_or(Error::<T>::CannotContractMigrator)?;
+                    Migrators::<T>::insert(migrator.clone(), new_migrations.clone());
+                    Self::deposit_event(RawEvent::MigratorContracted(migrator, new_migrations));
+                    Ok(())
+                },
+                None => fail!(Error::<T>::UnknownMigrator)
+            }
```

Comment:
```suggestion
            let new_migrations = Self::migrators(&migrator)
                .ok_or(Error::<T>::UnknownMigrator)?
                .checked_sub(decrease_migrations_by)
                .ok_or(Error::<T>::CannotContractMigrator)?;
            Migrators::<T>::insert(&migrator, &new_migrations);
            Self::deposit_event(RawEvent::MigratorContracted(migrator, new_migrations));
            Ok(())
```

Is this sort of comment valuable to you? Feel free to mark as resolved.

Reply:
Yes.